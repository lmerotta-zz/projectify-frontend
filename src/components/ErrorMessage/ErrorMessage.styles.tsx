import tw from "twin.macro";
import { motion } from "framer-motion";

export const Wrapper = motion(tw.span`text-red-600 text-sm`);
