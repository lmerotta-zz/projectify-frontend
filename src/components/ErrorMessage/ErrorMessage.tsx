import * as Styles from "./ErrorMessage.styles";
import { ErrorMessage as FormErrorMessage } from "@hookform/error-message";

type ErrorMessageProps = {
  name: string;
};
const ErrorMessage = ({ name }: ErrorMessageProps) => (
  <FormErrorMessage
    name={name}
    render={({ message }) => (
      <Styles.Wrapper data-testid={`error-message-${name}`} layout>
        {message}
      </Styles.Wrapper>
    )}
  />
);

export default ErrorMessage;
