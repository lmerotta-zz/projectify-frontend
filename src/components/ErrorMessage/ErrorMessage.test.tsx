import { render } from "@testing-library/react";
import { useEffect } from "react";
import { useForm, FormProvider } from "react-hook-form";
import ErrorMessage from "./ErrorMessage";

describe("ErrorMessage unit tests", () => {
  it("Displays the error message", async () => {
    const Component = () => {
      const methods = useForm();

      useEffect(() => {
        methods.setError("my-input", { message: "error-message" });
        // eslint-disable-next-line react-hooks/exhaustive-deps
      }, []);

      return (
        <FormProvider {...methods}>
          <ErrorMessage name="my-input" />
        </FormProvider>
      );
    };
    const { asFragment } = render(<Component />);

    expect(asFragment()).toMatchSnapshot();
  });
});
