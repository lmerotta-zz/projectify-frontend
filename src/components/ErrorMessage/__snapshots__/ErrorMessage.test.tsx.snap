// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage unit tests Displays the error message 1`] = `
<DocumentFragment>
  .emotion-0 {
  --tw-text-opacity: 1;
  color: rgba(220, 38, 38, var(--tw-text-opacity));
  font-size: 0.875rem;
  line-height: 1.25rem;
}

<span
    class="emotion-0"
    data-testid="error-message-my-input"
  >
    error-message
  </span>
</DocumentFragment>
`;
